src/partisan_hiscamp_membership_strategy.erl

```diff
@@ -61,26 +62,23 @@ join(#hiScamp{membership=Membership0, level1=L1, level2=L2}=State0, Node, _NodeS
     MembershipList = membership_list(State),
     {ok, Membership, OutgoingMessages, State}.
 
-try_distance(#hiScamp{membership=Membership0, distance=Dist}=_State0, Node) ->
-    Dist = 0,
+try_distance(#hiScamp{membership=Membership0, distance=Dist}=_State0, Node, Dict) ->
     Dict = get(distance_metrics),
-    try dict:find(Node, Dict) of
-        {ok, Dist} -> Dist
-    catch
-        {error, Error} -> Error
-    end.
+    case dict:find(Node, Dict) == error of
+        true ->
+            erlang:error("Cannot attain node distain");
+        false ->
```

Comment:
This is still wrong.

You need to pattern match on the *return* value of dict:find, otherwise, you won't get the value from the dictionary back.  In your example, you're calling dict:find, checking whether or not it's error, and then throwing the return value away.

Reply:
Ok obviously I'm misunderstanding something. How is the value being thrown away if Dist is being recorded and I'm updating Dist?
<img width="754" alt="screen shot 2018-12-17 at 1 37 17 pm" src="https://user-images.githubusercontent.com/27083906/50107531-fbd7f300-0200-11e9-9615-a2ecf5f5469a.png">

Reply:
You can't update dist this way. 

Remember, variables are single assignment and copied by value.  ```=``` isn't assignment, it's matching.  Therefore L73 will *crash* the system is the value is different than what it was defined as on L65.  If you want to change something, you have to return it to the caller who is passing the value in.